### ISSUE TITLE
static file routes are not ignored when using Flask>2.0

### ISSUE THREAD

**Issue by nicobrandt** [alisaifee/flask-limiter#285]
The comparison in https://github.com/alisaifee/flask-limiter/blob/master/flask_limiter/extension.py#L511

```python
        if (
            not request.endpoint
            or not (self.enabled and self.initialized)
            or view_func == current_app.send_static_file
            or name in self._exempt_routes
            or request.blueprint in self._blueprint_exempt
            or any(fn() for fn in self._request_filters)
            or g.get("_rate_limiting_complete")
        ):
            return
```

seems to be broken when using Flask-Limiter in combination with Flask>2.0.

Seems to be due to a change on how the static endpoint is registered in https://github.com/pallets/flask/blob/2.0.1/src/flask/app.py#L518

```python
            self_ref = weakref.ref(self)
            self.add_url_rule(
                f"{self.static_url_path}/<path:filename>",
                endpoint="static",
                host=static_host,
                view_func=lambda **kw: self_ref().send_static_file(**kw),  # type: ignore # noqa: B950
            )
```

as compared to https://github.com/pallets/flask/blob/1.1.4/src/flask/app.py#L605.

```python
            self.add_url_rule(
                self.static_url_path + "/<path:filename>",
                endpoint="static",
                host=static_host,
                view_func=self.send_static_file,
            )
```

Not quite sure how to fix it at the moment.

**Comment by nicobrandt** [comment]
FWIW, I ended up just using the following workaround for now, which works for my case:

```
@limiter.request_filter
def _endpoint_whitelist():
    return request.endpoint == "static"
```

Not sure if that would be a valid fix in the library as well, if someone could comment on that I can open a PR.

**Comment by alisaifee** [comment]
Thanks for the suggestion. The workaround seems legitimate and I'll add it in to master. This should be available soon in a 2.x release.

**Comment by alisaifee** [comment]
Addressed with 82bcc287c5df187b9058cdd608821201a5d245b1

**Comment by alisaifee** [comment]
Available in 2.0.0
